pkg/util/util.go

```diff
@@ -43,3 +45,18 @@ func ReadMTUConfigMap(ctx context.Context, client cnoclient.Client) (int, error)
 	klog.V(2).Infof("Found mtu %d", mtu)
 	return mtu, nil
 }
+
+// IsStandaloneManagedCluster returns true if the operator is running in a managed cluster that isn't managed by HyperShift.
+// It checks for the existence of the dedicated-admin namespace.
+func IsStandaloneManagedCluster(ctx context.Context, client cnoclient.Client) (bool, error) {
+	// TODO(martinkennelly): replace detection of a standalone managed cluster with a metric instead of a namespace when that metric
+	// becomes available.
+	err := client.Default().CRClient().Get(ctx, types.NamespacedName{Name: STANDALONE_MANAGED_CLUSTER_NAMESPACE}, &corev1.Namespace{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
```

Comment:
Shouldn't we just set this information somewhere in InfraStatus() which already handles a context (however appropriate is or isn't the actual context that is being used there)?

infraStatus is already being used in the relevant places and will drive where you can do the validation.
For cluster config, it fits in `ValidateClusterConfig`.
For oper config, it looks more like it would go into `IsChangeSafe`. It looks like `Validate` here is more about performing a static validation of the oper config itself, naming conventions notwithstanding.

Reply:
Thanks - It makes sense. Pat suggested the same earlier.
I agree the current intent for `Validate` is ensuring the values are within bounds and not checking if a value is ok wrt external (to the field) constraints (cluster env / supported).